app/lib/features/space/widgets/edit_space_sheet.dart

```diff
@@ -190,96 +165,51 @@ class _EditSpacePageConsumerState extends ConsumerState<EditSpacePage> {
           ],
         ),
       ),
-      actions: [
-        ElevatedButton(
-          onPressed: () => context.canPop()
-              ? context.pop()
-              : context.goNamed(Routes.main.name),
-          style: ElevatedButton.styleFrom(
-            backgroundColor: Theme.of(context).colorScheme.neutral,
-            shape: RoundedRectangleBorder(
-              borderRadius: BorderRadius.circular(6),
-            ),
-            side: BorderSide(
-              color: Theme.of(context).colorScheme.success,
-            ),
-            foregroundColor: Theme.of(context).colorScheme.neutral6,
-            textStyle: Theme.of(context).textTheme.bodySmall,
-          ),
-          child: const Text('Cancel'),
-        ),
-        const SizedBox(width: 10),
-        ElevatedButton(
-          onPressed: () async {
-            if (titleInput.isEmpty) {
-              customMsgSnackbar(
-                context,
-                'Please enter space name',
-              );
-              return;
-            }
-            // check permissions before updating space
-            bool havePermission = await permissionCheck();
-            // We are doing as expected, but the lints triggers.
-            // ignore: use_build_context_synchronously
-            if (!context.mounted) {
-              return;
-            }
-            if (!havePermission) {
-              showAdaptiveDialog(
-                context: context,
-                builder: (context) => DefaultDialog(
-                  title: Padding(
-                    padding: const EdgeInsets.all(8.0),
-                    child: Icon(
-                      Atlas.block_prohibited,
-                      size: 28,
-                      color: Theme.of(context).colorScheme.onError,
+      confirmActionTitle: 'Save Changes',
+      cancelActionTitle: 'Cancel',
+      confirmActionOnPressed: titleInput.trim().isEmpty
+          ? null
+          : () async {
+              // check permissions before updating space
+              bool havePermission = await permissionCheck();
+              if (!havePermission) {
+                // ignore: use_build_context_synchronously
+                customMsgSnackbar(
+                  context,
+                  'Cannot edit space with no permissions',
+                );
```

Comment:
shouldn't we return here?
```suggestion
                );
                return;
```